driver-core/src/main/com/mongodb/operation/BatchCursor.java

```diff
@@ -94,4 +95,18 @@
      * @return ServerAddress
      */
     ServerAddress getServerAddress();
+
+    /**
+     * Returns the resume token
+     *
+     * @return the resume token
+     */
+    BsonDocument getResumeToken();
```

Comment:
Can we just add these specializations to `ChangeStreamBatchCursor` rather than all cursors?

Reply:
From the spec, drivers must expose the resume token (https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#id21), and `MongoCursor` needs to access the resume token. Is there a way for `MongoCursor` to access the resume token in `ChangeStreamBatchCursor` and bypass the need for these methods at this level?

As for `getPostBatchResumeToken`, its only purpose is for the prose tests, which may explain why the prose tests did not exist before. The prose tests need access to the `postBatchResumeToken` to check that the resume token is initialized to it before any iteration occurs on the batch.

Reply:
My apologies I didn't realise the spec had been updated to stipulate that. 

In that case we could either continue as now and put it in all `MongoCursor` instances or for change streams override what the iterable returns and return an extended MongoCursor with those methods.

Reply:
I have created a new class named `MongoChangeStreamCursor` that extends `MongoBatchCursorAdapter` which contains the methods `getResumeToken()` and `getPostBatchResumeToken()`.

Reply:
:+1: